### ISSUE TITLE
artifacts-centos7-test: Failed with `Invalid URL '': No scheme supplied. Perhaps you meant https://?`

### ISSUE THREAD

**Issue by yaronkaikov** [scylladb/scylla-cluster-tests#6868]
It seems something general since it also `artifacts-ubuntu2204-test` failed with the same error

https://jenkins.scylladb.com/job/scylla-master/job/artifacts-offline-install/job/artifacts-centos7-test/353/console

```
07:07:33  < t:2023-11-28 05:07:33,812 f:decorators.py   l:75   c:sdcm.utils.decorators p:ERROR > 'get_url_content': Number of retries exceeded!
07:07:33  < t:2023-11-28 05:07:33,813 f:log.py          l:23   c:sdcm.utils.log       p:ERROR > Uncaught exception < t:2023-11-28 05:07:33,813 f:log.py          l:23   c:sdcm.utils.log       p:ERROR > Uncaught exception
07:07:33  < t:2023-11-28 05:07:33,813 f:log.py          l:23   c:sdcm.utils.log       p:ERROR > Traceback (most recent call last):
07:07:33  < t:2023-11-28 05:07:33,813 f:log.py          l:23   c:sdcm.utils.log       p:ERROR >   File "/tmp/workspace/scylla-master/artifacts-offline-install/artifacts-centos7-test/scylla-cluster-tests/./sct.py", line 1765, in <module>
07:07:33  < t:2023-11-28 05:07:33,813 f:log.py          l:23   c:sdcm.utils.log       p:ERROR >     cli.main(prog_name="hydra")
07:07:33  < t:2023-11-28 05:07:33,813 f:log.py          l:23   c:sdcm.utils.log       p:ERROR >   File "/usr/local/lib/python3.10/site-packages/click/core.py", line 1055, in main
07:07:33  < t:2023-11-28 05:07:33,813 f:log.py          l:23   c:sdcm.utils.log       p:ERROR >     rv = self.invoke(ctx)
07:07:33  < t:2023-11-28 05:07:33,813 f:log.py          l:23   c:sdcm.utils.log       p:ERROR >   File "/usr/local/lib/python3.10/site-packages/click/core.py", line 1657, in invoke
07:07:33  < t:2023-11-28 05:07:33,813 f:log.py          l:23   c:sdcm.utils.log       p:ERROR >     return _process_result(sub_ctx.command.invoke(sub_ctx))
07:07:33  < t:2023-11-28 05:07:33,813 f:log.py          l:23   c:sdcm.utils.log       p:ERROR >   File "/usr/local/lib/python3.10/site-packages/click/core.py", line 1404, in invoke
07:07:33  < t:2023-11-28 05:07:33,813 f:log.py          l:23   c:sdcm.utils.log       p:ERROR >     return ctx.invoke(self.callback, **ctx.params)
07:07:33  < t:2023-11-28 05:07:33,813 f:log.py          l:23   c:sdcm.utils.log       p:ERROR >   File "/usr/local/lib/python3.10/site-packages/click/core.py", line 760, in invoke
07:07:33  < t:2023-11-28 05:07:33,813 f:log.py          l:23   c:sdcm.utils.log       p:ERROR >     return __callback(*args, **kwargs)
07:07:33  < t:2023-11-28 05:07:33,813 f:log.py          l:23   c:sdcm.utils.log       p:ERROR >   File "/tmp/workspace/scylla-master/artifacts-offline-install/artifacts-centos7-test/scylla-cluster-tests/./sct.py", line 1126, in run_test
07:07:33  < t:2023-11-28 05:07:33,813 f:log.py          l:23   c:sdcm.utils.log       p:ERROR >     unittest.main(module=None, argv=['python -m unittest', argv],
07:07:33  < t:2023-11-28 05:07:33,813 f:log.py          l:23   c:sdcm.utils.log       p:ERROR >   File "/usr/local/lib/python3.10/unittest/main.py", line 100, in __init__
07:07:33  < t:2023-11-28 05:07:33,813 f:log.py          l:23   c:sdcm.utils.log       p:ERROR >     self.parseArgs(argv)
07:07:33  < t:2023-11-28 05:07:33,813 f:log.py          l:23   c:sdcm.utils.log       p:ERROR >   File "/usr/local/lib/python3.10/unittest/main.py", line 147, in parseArgs
07:07:33  < t:2023-11-28 05:07:33,813 f:log.py          l:23   c:sdcm.utils.log       p:ERROR >     self.createTests()
07:07:33  < t:2023-11-28 05:07:33,813 f:log.py          l:23   c:sdcm.utils.log       p:ERROR >   File "/usr/local/lib/python3.10/unittest/main.py", line 158, in createTests
07:07:33  < t:2023-11-28 05:07:33,813 f:log.py          l:23   c:sdcm.utils.log       p:ERROR >     self.test = self.testLoader.loadTestsFromNames(self.testNames,
07:07:33  < t:2023-11-28 05:07:33,813 f:log.py          l:23   c:sdcm.utils.log       p:ERROR >   File "/usr/local/lib/python3.10/unittest/loader.py", line 220, in loadTestsFromNames
07:07:33  < t:2023-11-28 05:07:33,813 f:log.py          l:23   c:sdcm.utils.log       p:ERROR >     suites = [self.loadTestsFromName(name, module) for name in names]
07:07:33  < t:2023-11-28 05:07:33,813 f:log.py          l:23   c:sdcm.utils.log       p:ERROR >   File "/usr/local/lib/python3.10/unittest/loader.py", line 220, in <listcomp>
07:07:33  < t:2023-11-28 05:07:33,813 f:log.py          l:23   c:sdcm.utils.log       p:ERROR >     suites = [self.loadTestsFromName(name, module) for name in names]
07:07:33  < t:2023-11-28 05:07:33,813 f:log.py          l:23   c:sdcm.utils.log       p:ERROR >   File "/usr/local/lib/python3.10/unittest/loader.py", line 191, in loadTestsFromName
07:07:33  < t:2023-11-28 05:07:33,813 f:log.py          l:23   c:sdcm.utils.log       p:ERROR >     return self.loadTestsFromModule(obj)
07:07:33  < t:2023-11-28 05:07:33,813 f:log.py          l:23   c:sdcm.utils.log       p:ERROR >   File "/usr/local/lib/python3.10/unittest/loader.py", line 124, in loadTestsFromModule
07:07:33  < t:2023-11-28 05:07:33,813 f:log.py          l:23   c:sdcm.utils.log       p:ERROR >     tests.append(self.loadTestsFromTestCase(obj))
07:07:33  < t:2023-11-28 05:07:33,813 f:log.py          l:23   c:sdcm.utils.log       p:ERROR >   File "/usr/local/lib/python3.10/unittest/loader.py", line 93, in loadTestsFromTestCase
07:07:33  < t:2023-11-28 05:07:33,813 f:log.py          l:23   c:sdcm.utils.log       p:ERROR >     loaded_suite = self.suiteClass(map(testCaseClass, testCaseNames))
07:07:33  < t:2023-11-28 05:07:33,813 f:log.py          l:23   c:sdcm.utils.log       p:ERROR >   File "/usr/local/lib/python3.10/unittest/suite.py", line 24, in __init__
07:07:33  < t:2023-11-28 05:07:33,813 f:log.py          l:23   c:sdcm.utils.log       p:ERROR >     self.addTests(tests)
07:07:33  < t:2023-11-28 05:07:33,813 f:log.py          l:23   c:sdcm.utils.log       p:ERROR >   File "/usr/local/lib/python3.10/unittest/suite.py", line 57, in addTests
07:07:33  < t:2023-11-28 05:07:33,813 f:log.py          l:23   c:sdcm.utils.log       p:ERROR >     for test in tests:
07:07:33  < t:2023-11-28 05:07:33,813 f:log.py          l:23   c:sdcm.utils.log       p:ERROR >   File "/tmp/workspace/scylla-master/artifacts-offline-install/artifacts-centos7-test/scylla-cluster-tests/sdcm/tester.py", line 296, in __init__
07:07:33  < t:2023-11-28 05:07:33,813 f:log.py          l:23   c:sdcm.utils.log       p:ERROR >     self._init_params()
07:07:33  < t:2023-11-28 05:07:33,813 f:log.py          l:23   c:sdcm.utils.log       p:ERROR >   File "/tmp/workspace/scylla-master/artifacts-offline-install/artifacts-centos7-test/scylla-cluster-tests/sdcm/tester.py", line 696, in _init_params
07:07:33  < t:2023-11-28 05:07:33,813 f:log.py          l:23   c:sdcm.utils.log       p:ERROR >     self.params = init_and_verify_sct_config()
07:07:33  < t:2023-11-28 05:07:33,813 f:log.py          l:23   c:sdcm.utils.log       p:ERROR >   File "/tmp/workspace/scylla-master/artifacts-offline-install/artifacts-centos7-test/scylla-cluster-tests/sdcm/sct_config.py", line 2469, in init_and_verify_sct_config
07:07:33  < t:2023-11-28 05:07:33,813 f:log.py          l:23   c:sdcm.utils.log       p:ERROR >     sct_config.get_version_based_on_conf()
07:07:33  < t:2023-11-28 05:07:33,813 f:log.py          l:23   c:sdcm.utils.log       p:ERROR >   File "/tmp/workspace/scylla-master/artifacts-offline-install/artifacts-centos7-test/scylla-cluster-tests/sdcm/sct_config.py", line 2330, in get_version_based_on_conf
07:07:33  < t:2023-11-28 05:07:33,813 f:log.py          l:23   c:sdcm.utils.log       p:ERROR >     scylla_version = get_branch_version(scylla_repo)
07:07:33  < t:2023-11-28 05:07:33,813 f:log.py          l:23   c:sdcm.utils.log       p:ERROR >   File "/tmp/workspace/scylla-master/artifacts-offline-install/artifacts-centos7-test/scylla-cluster-tests/sdcm/utils/version_utils.py", line 373, in get_branch_version
07:07:33  < t:2023-11-28 05:07:33,813 f:log.py          l:23   c:sdcm.utils.log       p:ERROR >     repo_details = get_repository_details(url=url)
07:07:33  < t:2023-11-28 05:07:33,813 f:log.py          l:23   c:sdcm.utils.log       p:ERROR >   File "/tmp/workspace/scylla-master/artifacts-offline-install/artifacts-centos7-test/scylla-cluster-tests/sdcm/utils/version_utils.py", line 361, in get_repository_details
07:07:33  < t:2023-11-28 05:07:33,813 f:log.py          l:23   c:sdcm.utils.log       p:ERROR >     urls = list({*get_url_content(url=url)})
07:07:33  < t:2023-11-28 05:07:33,813 f:log.py          l:23   c:sdcm.utils.log       p:ERROR >   File "/tmp/workspace/scylla-master/artifacts-offline-install/artifacts-centos7-test/scylla-cluster-tests/sdcm/utils/decorators.py", line 70, in inner
07:07:33  < t:2023-11-28 05:07:33,813 f:log.py          l:23   c:sdcm.utils.log       p:ERROR >     return func(*args, **kwargs)
07:07:33  < t:2023-11-28 05:07:33,813 f:log.py          l:23   c:sdcm.utils.log       p:ERROR >   File "/tmp/workspace/scylla-master/artifacts-offline-install/artifacts-centos7-test/scylla-cluster-tests/sdcm/utils/version_utils.py", line 258, in get_url_content
07:07:33  < t:2023-11-28 05:07:33,813 f:log.py          l:23   c:sdcm.utils.log       p:ERROR >     response = requests.get(url=url)
07:07:33  < t:2023-11-28 05:07:33,813 f:log.py          l:23   c:sdcm.utils.log       p:ERROR >   File "/usr/local/lib/python3.10/site-packages/requests/api.py", line 73, in get
07:07:33  < t:2023-11-28 05:07:33,813 f:log.py          l:23   c:sdcm.utils.log       p:ERROR >     return request("get", url, params=params, **kwargs)
07:07:33  < t:2023-11-28 05:07:33,813 f:log.py          l:23   c:sdcm.utils.log       p:ERROR >   File "/usr/local/lib/python3.10/site-packages/requests/api.py", line 59, in request
07:07:33  < t:2023-11-28 05:07:33,813 f:log.py          l:23   c:sdcm.utils.log       p:ERROR >     return session.request(method=method, url=url, **kwargs)
07:07:33  < t:2023-11-28 05:07:33,813 f:log.py          l:23   c:sdcm.utils.log       p:ERROR >   File "/usr/local/lib/python3.10/site-packages/requests/sessions.py", line 575, in request
07:07:33  < t:2023-11-28 05:07:33,813 f:log.py          l:23   c:sdcm.utils.log       p:ERROR >     prep = self.prepare_request(req)
07:07:33  < t:2023-11-28 05:07:33,813 f:log.py          l:23   c:sdcm.utils.log       p:ERROR >   File "/usr/local/lib/python3.10/site-packages/requests/sessions.py", line 486, in prepare_request
07:07:33  < t:2023-11-28 05:07:33,813 f:log.py          l:23   c:sdcm.utils.log       p:ERROR >     p.prepare(
07:07:33  < t:2023-11-28 05:07:33,813 f:log.py          l:23   c:sdcm.utils.log       p:ERROR >   File "/usr/local/lib/python3.10/site-packages/requests/models.py", line 368, in prepare
07:07:33  < t:2023-11-28 05:07:33,813 f:log.py          l:23   c:sdcm.utils.log       p:ERROR >     self.prepare_url(url, params)
07:07:33  < t:2023-11-28 05:07:33,813 f:log.py          l:23   c:sdcm.utils.log       p:ERROR >   File "/usr/local/lib/python3.10/site-packages/requests/models.py", line 439, in prepare_url
07:07:33  < t:2023-11-28 05:07:33,813 f:log.py          l:23   c:sdcm.utils.log       p:ERROR >     raise MissingSchema(
07:07:33  < t:2023-11-28 05:07:33,813 f:log.py          l:23   c:sdcm.utils.log       p:ERROR > requests.exceptions.MissingSchema: Invalid URL '': No scheme supplied. Perhaps you meant https://?
```

**Comment by vponomaryov** [comment]
Looks like recently merged https://github.com/scylladb/scylla-cluster-tests/pull/6832 PR causes this bug when `use_preinstalled_scylla=False` and `scylla_repo=''`.

@yaronkaikov try to set the `scylla_repo` and rerun.

**Comment by yaronkaikov** [comment]
> Looks like recently merged #6832 PR causes this bug when `use_preinstalled_scylla=False` and `scylla_repo=''`.
> 
> @yaronkaikov try to set the `scylla_repo` and rerun.

Still failing but took longer https://jenkins.scylladb.com/job/scylla-master/job/artifacts-offline-install/job/artifacts-centos7-test/356/

**Comment by yaronkaikov** [comment]
@fruch We now see it in 5.4 as well https://jenkins.scylladb.com/job/scylla-5.4/job/artifacts-offline-install/job/artifacts-centos7-test/12/console
